layers/+lang/elixir/packages.el

```diff
@@ -130,10 +130,15 @@
         (set (make-variable-buffer-local 'ruby-end-check-statement-modifiers)
              nil)
         (ruby-end-mode +1))
-      (add-hook 'elixir-mode-hook 'spacemacs//ruby-end))
+      (add-hook 'elixir-mode-hook 'spacemacs//ruby-end)
+      ;; hack to remove the autoloaded `add-hook' in `ruby-end'
+      ;; since they are inserted as an autoload, they have to be removed both
+      ;; before and after loading
```

Comment:
For my own understanding, why does them being inserted as an autoload necessitate removal in both config and init?